src/Integration/State/StateManager.cs

```diff
@@ -87,12 +89,16 @@ public void SetBoundProject(ProjectInformation project)
             ProjectViewModel projectViewModel = this.ManagedState.ConnectedServers.SelectMany(s => s.Projects).SingleOrDefault(p => p.ProjectInformation == project);
             Debug.Assert(projectViewModel != null, "Expecting a single project mapped to project information");
             this.ManagedState.SetBoundProject(projectViewModel);
+
+            this.BindingStateChanged?.Invoke(this, null);
```

Comment:
The normal pattern with event is to define On<EventName>(relevant arguments);
In this case OnBindingStateChange(){ 
this.BindingStateChange?.Inoke(this, EventArgs.Empty);
}